tests/sklearn/test_sklearn_models.py

```diff
@@ -1120,6 +1137,200 @@ def check_exposition_structural_methods_decision_trees(model, x, y):
+            )
+            cml_data_y_pred = loaded_from_data.predict_proba(
+                x,
+                fhe=mode,
+            )
+
+            # Compute accuracy
+            sklearn_accuracy = accuracy_score(sklearn_pred.argmax(axis=1), y)
+            cml_accuracy = accuracy_score(cml_y_pred.argmax(axis=1), y)
+            loaded_accuracy_from_threshold_accuracy = accuracy_score(
+                cml_threshold_y_pred.argmax(axis=1), y
+            )
+            loaded_accuracy_from_data_accuracy = accuracy_score(cml_data_y_pred.argmax(axis=1), y)
+
+            # Compare with sklearn
+            with subtests.test(
+                msg="Classifier Sklearn vs Threshold", n_bits=n_bits, tolerance=sklearn_tolerance
+            ):
+                value = numpy.abs(loaded_accuracy_from_threshold_accuracy - sklearn_accuracy)
+                assert (
+                    value < sklearn_tolerance
+                ), f"{loaded_accuracy_from_threshold_accuracy=} != {sklearn_accuracy} ({value})"
+            with subtests.test(
+                msg="Classifier Sklearn vs Data", n_bits=n_bits, tolerance=sklearn_tolerance
+            ):
+                value = numpy.abs(loaded_accuracy_from_data_accuracy - sklearn_accuracy)
+                assert (
+                    value < sklearn_tolerance
+                ), f"{loaded_accuracy_from_data_accuracy=} != {sklearn_accuracy} ({value})"
+
+            # Compare with CML final metric
+            with subtests.test(
+                msg="Classifier CML vs Threshold", n_bits=n_bits, tolerance=cml_tolerance
+            ):
+                value = numpy.abs(loaded_accuracy_from_threshold_accuracy - cml_accuracy)
+                assert (
+                    value < cml_tolerance
+                ), f"{loaded_accuracy_from_threshold_accuracy=} != {cml_accuracy} ({value})"
+            with subtests.test(
+                msg="Classifier CML vs Data", n_bits=n_bits, tolerance=cml_tolerance
+            ):
+                value = numpy.abs(loaded_accuracy_from_data_accuracy - cml_accuracy)
+                assert (
+                    value < cml_tolerance
+                ), f"{loaded_accuracy_from_data_accuracy=} != {cml_accuracy} ({value})"
+
+    # Regressor
+    else:
+        for n_bits, cml_tolerance, sklearn_tolerance in [
+            (max_n_bits, 0.8, 1e-5),
+            (reasonable_n_bits, 1.8, 1.8),
+        ]:
+            # Load a Concrete ML model from the fitted scikit-learn one
+            loaded_from_threshold = model_class.from_sklearn_model(
+                sklearn_model,
+                n_bits=n_bits,
+            )
+
+            loaded_from_data = model_class.from_sklearn_model(
+                sklearn_model,
+                X=x,
+                n_bits=n_bits,
+            )
+
+            # Compile both the initial Concrete ML model and the loaded one
+            concrete_model.compile(x)
+            mode = "disable"
+            if n_bits <= 8:
```

Comment:
we have constants like `N_BITS_THRESHOLD_FOR_CRT_FHE_CIRCUITS` in our tests if that's what you had in mind here

Reply:
Actually that won't really work with trees since we have rounding.

So I don't think it's possible from parameters only to known if the circuit will use the CRT representation.

Reply:
Ho yeah I should use `reasonable_n_bits` here